### PR TITLE
Add custom column example for existing Kind

### PR DESCRIPTION
**Pull request by jescalada**
## Hey, I just made a Pull Request!

Related Issue: #27238

**Note:** I might have completely missed the point with my example. Should the user only be able to modify the `App.tsx`, in which case we want to pass in a `CatalogTableColumnsFunc` which checks for the User entity and adds the columns instead of an array to the `CatalogIndexPage` component? Or is what I wrote a valid approach too? In both cases the desired behaviour can be achieved.

Implemented User Email column based on example code:
image

### Changelog
- Added new examples to `software-catalog/catalog-customization.md` doc on how to create or modify columns for existing Kinds.
- Clarified/reworded existing sections

### CODE REVIEW

**Review by vinzscam — CHANGES_REQUESTED** [review]
Thank you for your work on this, @jescalada 🙏
Would it be possible to adapt this procedure to a Backstage app layout created using npx @backstage/create-app? This is the recommended approach for creating Backstage apps, and users will not have a /plugins/catalog/* folder.

**Comment by jescalada** [comment]
> Thank you for your work on this, @jescalada 🙏 Would it be possible to adapt this procedure to a Backstage app layout created using npx @backstage/create-app? This is the recommended approach for creating Backstage apps, and users will not have a /plugins/catalog/* folder.

I suspected this was the case 😅 Thankfully it only required redefining the email column definition on the `App.tsx`! Now the whole example only requires editing the `App.tsx` file.

File: docs/features/software-catalog/catalog-customization.md [hunk 1]
```diff
@@ -63,7 +63,92 @@ There are many options that can be set using `tableOptions`, the full list of se
...
+{
+  /* highlight-add-start */
+}
+const myColumnsFunc: CatalogTableColumnsFunc = entityListContext => {
+  if (entityListContext.filters.kind?.value === 'user') {
+    return [
+      // Existing columns (Name, Description, Tags)
+      CatalogTable.columns.createNameColumn(),
+      CatalogTable.columns.createMetadataDescriptionColumn(),
+      CatalogTable.columns.createTagsColumn(),
```
Comment (vinzscam) [inline docs/features/software-catalog/catalog-customization.md:82]:
```suggestion
    return [
      // Render existing columns
      ...CatalogTable.defaultColumnsFunc(entityListContext),
```

File: docs/features/software-catalog/catalog-customization.md [hunk 2]
```diff
@@ -63,7 +63,92 @@ There are many options that can be set using `tableOptions`, the full list of se
...
+
+  return CatalogTable.defaultColumnsFunc(entityListContext);
+};
+{
+  /* highlight-add-end */
+}
+```
+
+Some other possible values for the existing entity kinds are `user`, `domain`, `system`, `group`, `template` and `location`.
+
```
Comment (vinzscam) [inline docs/features/software-catalog/catalog-customization.md:95]: I'm thinking we could skip this

```suggestion
```

File: docs/features/software-catalog/catalog-customization.md [hunk 3]
```diff
@@ -63,7 +63,92 @@ There are many options that can be set using `tableOptions`, the full list of se
...
+});
+{/* highlight-add-end */}
+
+const myColumnsFunc: CatalogTableColumnsFunc = entityListContext => {
+  if (entityListContext.filters.kind?.value === 'user') {
+    return [
+      // Existing columns (Name, Description, Tags)
+      CatalogTable.columns.createNameColumn(),
+      CatalogTable.columns.createMetadataDescriptionColumn(),
+      CatalogTable.columns.createTagsColumn(),
```
Comment (vinzscam) [inline docs/features/software-catalog/catalog-customization.md:118]:
```suggestion
    return [
      // Render existing columns
      ...CatalogTable.defaultColumnsFunc(entityListContext),
```

**Review by vinzscam — CHANGES_REQUESTED** [review]
this looks great @jescalada 🙏

added just some minor suggestions

**Comment by jescalada** [comment]
Thanks for the review! I committed the requested changes.

**Comment by jescalada** [comment]
Pinging @vinzscam for visibility :)

**Review by awanlin — APPROVED** [review]
Thanks for adding this @jescalada 🚀

**Pull request merged**